### PR TITLE
FEAT: untar input directory if needed.

### PR DESCRIPTION
**Pull request by DriesSchaumont**

### CODE REVIEW

File: src/demultiplex/main.nf [hunk 1]
```diff
@@ -4,7 +4,18 @@ workflow run_wf {
 
   main:
     output_ch = input_ch
-
+      // untar input if needed
+      | untar.run(
+        runIf: {id, state -> 
+          state.input.toString().endsWith(".tar.gz") ? true : false
```
Comment (tverbeiren) [inline src/demultiplex/main.nf:10]: Or `.tar` or `.tgz`?

**Comment by tverbeiren** [comment]
Some comments about the last two commits:

1. `untar` did not deal with archives that do not start with `./`.
2. Writing the contents of a tar file to disk and `grep`'ing on that turns out to be twice as fast as before.
3. The `main.nf` file was a good start, but did not work the way it was intended. Partlyh because the `runIf` should result in a similar datastructure as when the module would not have run, i.e. `output` should be mapped to `input`.
4. I left a `view` that should be removed before actually merging.

The test was run with the test dataset from nf-core: 

```
id,samplesheet,lane,flowcell
35147139,https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/testdata/miseq_35147139/miseq_35147139_samplesheet.csv,1,s3://ngi-igenomes/test-data/demultiplex/miseq_35147139.tar.gz
```

**Comment by DriesSchaumont** [comment]
> Some comments about the last two commits:
> 
>     1. `untar` did not deal with archives that do not start with `./`.
> 
>     2. Writing the contents of a tar file to disk and `grep`'ing on that turns out to be twice as fast as before.
> 
>     3. The `main.nf` file was a good start, but did not work the way it was intended. Partlyh because the `runIf` should result in a similar datastructure as when the module would not have run, i.e. `output` should be mapped to `input`.
> 
>     4. I left a `view` that should be removed before actually merging.
> 
> 
> The test was run with the test dataset from nf-core:
> 
> ```
> id,samplesheet,lane,flowcell
> 35147139,https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/testdata/miseq_35147139/miseq_35147139_samplesheet.csv,1,s3://ngi-igenomes/test-data/demultiplex/miseq_35147139.tar.gz
> ```
For future reference:

1. Was fixed by #6 
2. True! At first this was debugging output I forgot to remove, but I want to keep it because it might be usefull later.
3. Doh! Should have tested it before creating the PR :)
4. Yes.

**Comment by DriesSchaumont** [comment]
I added some more updates:

- Created .gitignore
- Bumped viash
- Updated the test resources
- Merged updates from main into this PR

**Review by tverbeiren — APPROVED** [review]
🚀

**Pull request merged**